build_modules/lib/src/module_cache.dart

```diff
@@ -53,9 +60,24 @@ class DecodingCache<T> {
   /// content dependencies will be tracked through [reader].
   Future<T> find(AssetId id, AssetReader reader) async {
     if (!await reader.canRead(id)) return null;
-    var result = _cached.putIfAbsent(
-        id, () => Result.capture(reader.readAsBytes(id).then(_fromBytes)));
-    return Result.release(result);
+    if (_cached.containsKey(id)) {
+      final entry = _cached[id];
+      if (entry.needsCheck) {
+        final digest = await reader.digest(id);
```

Comment:
We might want to put the digest into a `Future<Result>` as well...

I think we could end up doing some duplicate work here if two builders ask for the same content that needs to be checked - both will try to get the digest and both will read and do the serialization...

I'll need to give this some thought.